contentcuration/contentcuration/frontend/channelEdit/components/modals/ResubmitChannelModal.vue

```diff
@@ -0,0 +1,77 @@
+<template>
+
+  <KModal
+    v-if="value"
+    :title="title"
+    :submitText="resubmitButtonText"
+    :cancelText="dismissButtonText"
+    data-test="resubmit-modal"
+    @submit="handleResubmit"
+    @cancel="handleDismiss"
+  >
+    <div class="resubmit-modal-content">
+      <p class="resubmit-modal-description">{{ description }}</p>
+      <p class="resubmit-modal-question">{{ question }}</p>
+    </div>
+  </KModal>
+
+</template>
+
+
+<script setup>
+
+  import { computed } from 'vue';
+  import { communityChannelsStrings } from 'shared/strings/communityChannelsStrings';
+
+  const props = defineProps({
+    value: {
+      type: Boolean,
+      default: false,
+    },
+    channel: {
+      type: Object,
+      required: true,
+      validator(value) {
+        return value && typeof value.name === 'string' && typeof value.version === 'number';
+      },
+    },
+    latestSubmissionVersion: {
+      type: Number,
+      default: null,
+    },
+  });
+
+  const emit = defineEmits(['dismiss', 'resubmit', 'input']);
```

Comment:
Oooh, similar to what we did for the DeleteModal [here](https://github.com/learningequality/studio/pull/5551#discussion_r2523746143), could we also use the `@close` event for the parent component to close the Modal instead of using `dismiss` and `v-model`? So we can keep this consistent pattern for rendering modals!